### PR TITLE
Make v1.2.0

### PR DESCRIPTION
**Pull request by EraYaN**
Move three fields into one "baseurl".
Add URL validation.
Put all CSS at the top of the page.

### CODE REVIEW

**Comment by dkanada** [comment]
@dmitrylyzo thoughts?

**Comment by dmitrylyzo** [comment]
I can only test in emulator.

If I use `BaseURL=/jellyfin` and not specify it in the field, I cannot abort connecting.
`VM75:1 Uncaught SyntaxError: Unexpected token < in JSON at position 0`

**Comment by EraYaN** [comment]
I that makes sense because it will request whatever is at your base domain and that probably sends back an HTML 404 page. It's can't just magically guess your jellyfin baseurl. Although if you enter your full domain in the main field like `https://jellyfin.example.com/jellyfin` it should work.

**Comment by dmitrylyzo** [comment]
In my case, it returns code 200 and an empty `responseText`.
404 error handled normally.

**Comment by EraYaN** [comment]
Well that depends on what you are also hosting on that server. No idea what your say nginx does, but it's definitely a config problem.

**Comment by Narfinger** [comment]
On my TV this currently does not show an error if jellyfin is unreachable and just returns to the login screen again without any message.

File: org.jellyfin.webos/index.html [hunk 1]
```diff
@@ -362,16 +355,14 @@
 			<img src="assets/banner-dark.png">
 		</div>
 		<div id="serverInfoForm">
-			<p style="display:none; color: red; font-weight: bold; text-align: center;" id="error">&nbsp;</p>
-			<form>
-				<input type="text" id="hostname" name="hostname" placeholder="jellfin.local" tabindex="0" /><br>
-				<input type="text" id="port" name="port" placeholder="8096" tabindex="1" /><br>
-				<input type="text" id="schema" name="schema" placeholder="https" tabindex="2" /><br>
+			<p id="error">&nbsp;</p>
```
Comment (dmitrylyzo) [inline org.jellyfin.webos/index.html]:
```suggestion
			<p id="error" style="display: none;">&nbsp;</p>
```

File: org.jellyfin.webos/index.html [hunk 2]
```diff
@@ -52,6 +52,17 @@
 			margin: 5px 1px 3px 1px;
 			border-radius: 1em;
 		}
+		
+		#baseurl {
+			width: 100%;
+		}
+		
+		#error {
+			display:none;
```
[the remark this answers is not in the document]
Comment (EraYaN) [reply org.jellyfin.webos/index.html]: The `displayError` function show set the style element on the error class and I believe that should override the global css. Can't I remember testing this though, but also not that is was broken.

**Comment by Narfinger** [comment]
New version fixes my comment from above. Good job!

**Comment by dkanada** [comment]
@EraYaN I didn't notice that this pull request conflicted with the other one, can you rebase the changes against the new file?

**Comment by JustAMan** [comment]
> I that makes sense because it will request whatever is at your base domain and that probably sends back an HTML 404 page. It's can't just magically guess your jellyfin baseurl. Although if you enter your full domain in the main field like `https://jellyfin.example.com/jellyfin` it should work.

Sounds more like json parsing errors are unhandled.
Also I think there should be a timeout or something preventing this from hanging.

**Pull request merged**